python/lsst/source/injection/utils/consolidate_injected_deepCoadd_catalogs.py

```diff
@@ -0,0 +1,624 @@
+):
+    """Set a flag for each source if it is in the innerBBox of a patch.
+
+    Parameters
+    ----------
+    catalog: `astropy.table.Table`
+        A catalog of sources.
+    patchInfo : `lsst.skymap.PatchInfo`
+        Information about a `SkyMap` `Patch`.
+    col_ra: `str`
+        Column name for right ascension (in degrees).
+    col_dec: `str`
+        Column name for declination (in degrees).
+
+    Returns
+    -------
+    isPatchInner : array-like of `bool`
+        `True` for each source that has a centroid
+        in the inner region of a patch.
+    """
+    # Extract positions for all the sources.
+    ra = catalog[col_ra]
+    dec = catalog[col_dec]
+
+    # convert the coordinates to pixel positions
+    wcs = patchInfo.getWcs()
+    x, y = wcs.skyToPixelArray(ra, dec, degrees=True)
+
+    # set inner flags for each source
+    innerFloatBBox = Box2D(patchInfo.getInnerBBox())
+    isPatchInner = innerFloatBBox.contains(x, y)
+
+    return isPatchInner
+
+
+def getTractInner(
+    catalog,
+    tractInfo,
+    skyMap,
+    col_ra: str = "ra",
+    col_dec: str = "dec",
+):
+    """Set a flag for each source that the skyMap includes in tractInfo.
+
+    Parameters
+    ----------
+    catalog: `astropy.table.Table`
+        A catalog of sources.
+    tractInfo : `lsst.skymap.TractInfo`
+        Tract object
+    skyMap : `lsst.skymap.BaseSkyMap`
+        Sky tessellation object
+    col_ra: `str`
+        Column name for right ascension (in degrees).
+    col_dec: `str`
+        Column name for declination (in degrees).
+
+    Returns
+    -------
+    isTractInner : array-like of `bool`
+        True if the skyMap.findTract method returns
+        the same tract as tractInfo.
+    """
+    ras, decs = catalog[col_ra].value, catalog[col_dec].value
+    skyCoords = [SpherePoint(ra, dec, degrees) for (ra, dec) in list(zip(ras, decs))]
+    tractId = tractInfo.getId()
+    isTractInner = np.array([skyMap.findTract(coord).getId() == tractId for coord in skyCoords])
+    return isTractInner
+
+
+def setPrimaryFlags(
+    catalog,
+    skyMap,
+    tractInfo,
+    patches: list,
+    col_ra: str = "ra",
+    col_dec: str = "dec",
+    isPatchInnerKey: str = "injected_isPatchInner",
+    isTractInnerKey: str = "injected_isTractInner",
+    isPrimaryKey: str = "injected_isPrimary",
+    injectionKey: str = "injection_flag",
+):
+    """Set isPrimary and related flags on sources.
+
+    For coadded imaging, the `isPrimary` flag returns True when an object
+    has no children, is in the inner region of a coadd patch, is in the
+    inner region of a coadd trach, and is not detected in a pseudo-filter
+    (e.g., a sky_object).
+    For single frame imaging, the isPrimary flag returns True when a
+    source has no children and is not a sky source.
+
+    Parameters
+    ----------
+    catalog: `astropy.table.Table`
+        A catalog of sources.
+        Writes is-patch-inner, is-tract-inner, and is-primary flags.
```

Comment:
Should these be - or _?

Reply:
I got these doc strings straight from the code in meas_algorithms, so I'd assume it probably doesn't matter

Reply:
As I read it, "is-patch-inner" is meant to be descriptive, rather than referencing anything specific in the code. I.e., "This thing writes an 'is-patch-inner' flag". Is that accurate?